File: daq/slc/src/dqm/src/Histo2I.cc
#include "dqm/Histo2I.h"

#include "dqm/IntArray.h"

using namespace Belle2;

Histo2I::Histo2I() : Histo2()
{
  _data = new IntArray();
}

Histo2I::Histo2I(const Histo2I& h) : Histo2(h)
{
  if (h._data != NULL) _data = new IntArray(*h._data);
  else _data = new IntArray();
}

Histo2I::Histo2I(const std::string& name, const std::string& title,
                 int nbinx, double xmin, double xmax,
                 int nbiny, double ymin, double ymax)
  : Histo2(name, title, nbinx, xmin, xmax, nbiny, ymin, ymax)
{
  _data = new IntArray((nbinx + 2) * (nbiny + 2));
}

std::string Histo2I::getDataType() const throw()
{
  return std::string("H2I");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/Histo2I.java
package b2daq.dqm.core;


public class Histo2I extends Histo2 {

	public Histo2I() {
		super();
		data = new IntArray(0);
	}
	public Histo2I(String name, String title, int nbinx, double xmin, double xmax,
			int nbiny, double ymin, double ymax) {
		super(name, title, nbinx, xmin, xmax, nbiny, ymin, ymax);
		data = new IntArray((nbinx+2)*(nbiny+2));
	}
	public Histo2I(Histo2I h) {
		super();
		data = new IntArray(0);
	}
	public String getDataType() {
		return "H2I";
	}

}
